src/rpc/validation/SafeValidator.ts

```diff
@@ -519,283 +494,131 @@ export class SafeValidator
         userOperation: UserOperationV07,
         entryPoint: Address
     ): Promise<[ValidationResultV07, BundlerTracerResult]> {
+        if (!this.entryPointSimulationsAddress) {
+            throw new Error("entryPointSimulationsAddress is not set")
+        }
+
         const packedUserOperation = toPackedUserOperation(userOperation)
 
         const entryPointSimulationsCallData = encodeFunctionData({
             abi: EntryPointV07SimulationsAbi,
-            functionName: "simulateValidation",
-            args: [packedUserOperation]
+            functionName: "simulateValidationLast",
+            args: [[packedUserOperation]]
         })
 
-        const callData = encodeDeployData({
+        const callData = encodeFunctionData({
             abi: PimlicoEntryPointSimulationsAbi,
-            bytecode: PimlicoEntryPointSimulationsBytecode,
-            args: [entryPoint, entryPointSimulationsCallData]
+            functionName: "simulateEntryPoint",
+            args: [entryPoint, [entryPointSimulationsCallData]]
         })
 
         const tracerResult = await debug_traceCall(
             this.publicClient,
             {
                 from: zeroAddress,
+                to: this.entryPointSimulationsAddress,
                 data: callData
             },
             {
                 tracer: bundlerCollectorTracer
             }
         )
 
-        const lastResult = tracerResult.calls.slice(-1)[0]
-        if (lastResult.type !== "REVERT") {
```

Comment:
Hey just to be sure we don't need this because we don't revert in the PimlicoSimulationsContract?

Reply:
good catch, re-added this check